platform/darwin/src/MGLConversionValue.h

```diff
@@ -0,0 +1,68 @@
+#include <string>
+
+class MGLConversionValue {
```

Comment:
What's the purpose of `MGLConversionValue` in this PR, as opposed to inlining each of its methods in the corresponding method in darwin_conversion.hpp? E.g.:

```objc
inline bool isUndefined(NSObject *value) {
    return !value || value == [NSNull null];
}
```